### ISSUE TITLE
Change final for mut keyword

### ISSUE THREAD

**Issue by JorgeCepeda** [manifold-systems/manifold#505]
I don't like the idiom of making most variables `final,` adding words to the code.

Instead I'd prefer to assume variables are final unless there's a `mut` keyword added to it, like in other languages, and remove the `final` keyword altogether.

**Comment by CC007** [comment]
Having a setting that can specify `final` being the default for fields, parameters and local vars would indeed be nice. 

You probably don't want to touch `final` methods and classes though, since it has a somewhat different meaning there.

**Comment by JorgeCepeda** [comment]
I agree, in the case of `final` classes I see it as a good option to favour composition over inheritance
